lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt

```diff
@@ -311,32 +360,120 @@ class JsonRpcProvider(
         return buildRequest(JsonRpcMethod.GET_EVENTS, params, GetEventsResult.serializer())
     }
 
-    private fun getEstimateFee(payload: EstimateFeePayload): Request<EstimateFeeResponse> {
-        val jsonPayload = Json.encodeToJsonElement(payload)
+    private fun getEstimateFee(payload: EstimateInvokeFunctionFeePayload): Request<EstimateFeeResponse> {
+        val jsonPayload = Json { encodeDefaults = true }.encodeToJsonElement(payload)
```

Comment:
I think we are reconstructing Json object in every method doing it like this. Perhaps extracting it to a class member would be better?

Reply:
Not sure I follow, can you elaborate a bit more?

Reply:
Something along the lines of having 
```kotlin
private val json... = Json { encodeDefaults = true }
```
in the class and then using it in the methods. As far as I understand, everytime you do `Json { ... }` a new object has to be created.

Reply:
Should this then be used everywhere? there are multiple places we use `Json.encodeToJsonElement` which I suppose is the same thing?

Reply:
So I tried to investigate this, and this is what I concluded:

I'm not sure about the details, but if try too look in the code of `Json` without any parameters is't actually a companion object named `Default` of Json class that is also an instance of Json class itself (didn't even know it's possible) using some sort of default configuration.

So I'm assuming anytime you use `Json` there's no performance penalty because there's a single default instance of `Json`. 

However when you do `Json { ... }` (again, look at the code, it's also confusing to me 😅), you're using a [sort of builder](https://kotlinlang.org/docs/type-safe-builders.html) you're creating new instances of `Json` with your configuration.

Reply:
🤯 

that's cool, thanks for the extra effort to check this. Changed in newest commit :+1: